### PR TITLE
Add access control logging

### PR DESCRIPTION
**Pull request by mlepage-google**
#### Problem
Need a bit of visibility into what access control is doing before
we turn on enforcement.

#### Change overview
Add a log statement in AccessControl::Check so we can see what
params are being used to do the check.

This looks like:
AccessControl: checking f=1 a=c s=0x0000000000BC5C01 t= c=0x0000_0030 e=0 p=o

With some CATs:
AccessControl: checking f=1 a=c s=0x0000000000BC5C01 t=0xAAAA0001,0xBBBB0002 c=0x0000_0030 e=0 p=o

Where AccessControl::Check is called, if there's an error,
override it for now (grace period) but log it as an error.

This looks like:
AccessControl: overriding DENY (for now)

Progress toward issue #14454

#### Testing
- Built all-clusters-app and ran on Linux with REPL
- Checked logs for expected output

### CODE REVIEW

File: src/access/AccessControl.cpp [hunk 1]
```diff
@@ -61,6 +61,77 @@ bool CheckRequestPrivilegeAgainstEntryPrivilege(Privilege requestPrivilege, Priv
...
+        return 'c';
+    case AuthMode::kGroup:
+        return 'g';
+    }
+    return 'u';
+}
+
+char * GetCatStringForLogging(char * buf, size_t size, const CATValues & cats)
+{
+    VerifyOrDie(size >= 16);
```
Comment (msandstedt) [inline src/access/AccessControl.cpp:82]: Seems like a good place to pass a FixedSpan if you want to know that the buffer will be a certain size.  And certainly so given there's only one call.

This is just a style nit though.

**Review by msandstedt — APPROVED** [review]

File: src/access/AccessControl.cpp [hunk 2]
```diff
@@ -88,6 +159,15 @@ CHIP_ERROR AccessControl::Check(const SubjectDescriptor & subjectDescriptor, con
     // Don't check if using default delegate (e.g. test code that isn't testing access control)
     ReturnErrorCodeIf(&mDelegate == &mDefaultDelegate, CHIP_NO_ERROR);
 
+    {
+        char buf[64]; // fits 5 cats
+        ChipLogDetail(DataManagement,
+                      "AccessControl: checking f=%" PRIu8 " a=%c s=0x%" PRIx64 " t=%s c=0x%" PRIx32 " e=%" PRIu16 " p=%c",
```
Comment (bzbarsky-apple) [inline src/access/AccessControl.cpp:165]: PRIx64 blows up on a bunch of the embedded libcs.  Please use ChipLogFormatX64/ChipLogValueX64.

**Review by bzbarsky-apple — APPROVED** [review]

File: src/access/AccessControl.cpp (hunk 2 above)
Comment (mlepage-google) [reply src/access/AccessControl.cpp:165]: Done, more verbose though. :-(

File: src/access/AccessControl.cpp [hunk 3]
```diff
@@ -88,6 +159,15 @@ CHIP_ERROR AccessControl::Check(const SubjectDescriptor & subjectDescriptor, con
...
+                      subjectDescriptor.fabricIndex, GetAuthModeStringForLogging(subjectDescriptor.authMode),
+                      subjectDescriptor.subject, GetCatStringForLogging(buf, sizeof(buf), subjectDescriptor.cats),
+                      requestPath.cluster, requestPath.endpoint, GetPrivilegeStringForLogging(requestPrivilege));
```
[the remark this answers is not in the document]
Comment (mlepage-google) [reply src/access/AccessControl.cpp:168]: Done, more verbose though. :-(

File: src/access/AccessControl.cpp (hunk 1 above)
Comment (mlepage-google) [reply src/access/AccessControl.cpp:82]: I'm seeing just one use of FixedSpan in the code, does that seem right?

Anyways, it doesn't have to be a fixed size, it just has to be a reasonable minimum, and will use more if provided.

**Review by andy31415 — APPROVED** [review]

File: src/access/AccessControl.cpp (hunk 1 above)
Comment (msandstedt) [reply src/access/AccessControl.cpp:82]: This is just a nit.   No need to add any changes.

My observation is essentially that there is a runtime assert here, when in fact we could have a compile-time check.

Comment (mlepage-google) [reply src/access/AccessControl.cpp]: I shuffled it slightly to remove the verify. (This was just a nicety to ensure we could put at least a couple characters, the slight tweaks make it handle even those cases.)

Comment (msandstedt) [reply src/access/AccessControl.cpp]: Thanks for the amendment.  This addresses my concern.

**Pull request merged**